src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs

```diff
@@ -514,22 +513,45 @@ public override int Add(object value)
         /// </summary>
         private sealed class HttpWebRequestArrayList : ArrayListWrapper
         {
-            public HttpWebRequestArrayList(ArrayList list) : base(list)
+            private readonly object _Connection;
+
+            public HttpWebRequestArrayList(object connection, ArrayList list) : base(list)
             {
+                _Connection = connection;
             }
 
             public override int Add(object value)
             {
+                // Add before firing events so if some user code cancels/aborts the request it will be found in the outstanding list.
+                int index = base.Add(value);
```

Comment:
If reverted `TestCanceledRequest` will fail because no Exception event is fired. To cancel the request it does this: `new EventObserverAndRecorder(_ => { cts.Cancel(); })`

Before:
1) m_WriteList.Add is called
2) Start event fired
3) User code called (cts.Cancel)
4) Request is aborted
5) Connection looks for an active request in m_WriteList, finds nothing (https://referencesource.microsoft.com/#system/net/System/Net/_Connection.cs,2816). If it finds something a lot more code happens including a potential call to m_WriteList.Clear
6) Request is added to the ArrayList. It might be leaked at this point, no cleanup will happen because abort/cleanup has already completed?

After:
1) m_WriteList.Add is called
2) Request is added to the ArrayList
3) Start event fired
4) User code called (cts.Cancel)
5) Request is aborted
6) Connection looks for a request to remove in m_WriteList, successfully finds the active one. Is able to do its cleanup which causes the Exception event to fire correctly